pkg/generators/git.go

```diff
@@ -174,29 +177,36 @@ func (g *GitGenerator) generateParamsFromGitFile(filePath string, fileContent []
 
 }
 
-func (g *GitGenerator) filterApps(Directories []argoprojiov1alpha1.GitDirectoryGeneratorItem, allPaths []string) []string {
+func (g *GitGenerator) filterPaths(pathType string, items []argoprojiov1alpha1.GitGeneratorItem, allPaths []string) []string {
 	res := []string{}
-	for _, appPath := range allPaths {
-		appInclude := false
-		appExclude := false
-		// Iterating over each appPath and check whether directories object has requestedPath that matches the appPath
-		for _, requestedPath := range Directories {
-			match, err := path.Match(requestedPath.Path, appPath)
-			if err != nil {
-				log.WithError(err).WithField("requestedPath", requestedPath).
-					WithField("appPath", appPath).Error("error while matching appPath to requestedPath")
-				continue
+	for _, itemPath := range allPaths {
+		include := false
+		exclude := false
+		for _, requestedPath := range items {
+			var match bool
+			var err error
+			if pathType == "files" {
+				pathGlob := glob.MustCompile(requestedPath.Path)
```

Comment:
If it's important to use glob instead of `path.Match`, I think we should use a `glob` field instead of the `path` field, to avoid confusion.

If it's not necessary to use glob, I'd recommend we just use `path.Match`.

Reply:
@crenshaw-dev, we need to use glob if the path contains "**", I will add a conditional that checks if the path contains that and then uses glob but falls back to path.Match if it does not.

Reply:
That is a cool trick... Could you apply it to the directory generator as well? I usually don't like implicit toggles, but I think if we add a line to the documentation about it, it'll be fine.

Reply:
Both the file and directory generator will use the same method for filtering so it will be applied to both.

Reply:
I have pushed the changes and updated the examples and docs